File: #9.py
def largest_nonadj_sum(arr):
	sum_incl = arr[0]
	sum_excl = 0

	for i in range(1, len(arr)):
		new_incl = sum_excl + arr[i]
		sum_excl = max(sum_incl, sum_excl)
		sum_incl = new_incl

	return max(sum_incl, sum_excl)


while True:
	arr = [int(x) for x in input().split(',')]
	print(largest_nonadj_sum(arr))

File: #9.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <sstream>
using namespace std;

int largestNonAdjacentSum(vector<int> &v) {
	int sumExcl = 0;
	int sumIncl = v[0];

	for (int i = 1; i < v.size(); i++) {
		int temp = max(sumExcl, sumIncl);
		sumIncl = sumExcl + v[i];
		sumExcl = temp;
	}

	return max(sumExcl, sumIncl);
}

int main() {
	string input;
	while (getline(cin, input)) {
		vector<int> v;
		istringstream ss(input);
		string token;
		while (getline(ss, token, ',')) {
			v.push_back(stoi(token));
		}
		cout << largestNonAdjacentSum(v) << endl;
	}
}